src/node/history.h

```diff
@@ -398,8 +399,14 @@ namespace ccf
     void add_result(
       kv::TxHistory::RequestID id,
       kv::Version version,
-      const std::vector<uint8_t>& data) override
+      const std::vector<uint8_t>& data_replicated,
+      const std::vector<uint8_t>& data_derived) override
     {
+      std::vector<uint8_t> data;
+      data.reserve(data_replicated.size() + data_derived.size());
```

Comment:
Minor: This could just be `std::vector<uint8_t> data(data_replicate.size() + data_derived.size());`

Reply:
Not in this case. If we set the size on initialization we get a vector that size but with default-initialized values populating the vector, where as if we use `reserve` it just reserves the memory but doesn't initialize it. This makes a difference since we are using `data.end()` when we do `data.insert()`.